applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -434,6 +431,10 @@ def _execute_after_reading(self):
         import check_and_prepare_model_process_structural
         check_and_prepare_model_process_structural.CheckAndPrepareModelProcess(self.main_model_part, params).Execute()
 
+        # This will be removed once the Model is fully supported! => It wont e necessary anymore
+        if not self.model.HasModelPart(self.main_model_part.Name):
```

Comment:
... and here

Can you explain pls?

Reply:
Here is added in the normal case that the restart is not used. Besides it is added just once, because the check is done before computing

Reply:
Ok now I finally understood
this is needed for the ReadMaterialProcess right?

Can you please add a comment abt this?

Reply:
Yes